turbopack/crates/turbopack-trace-server/src/reader/turbopack.rs

```diff
@@ -294,8 +296,15 @@ impl TurbopackFormat {
 }
 
 impl TraceFormat for TurbopackFormat {
-    fn read(&mut self, mut buffer: &[u8]) -> Result<usize> {
-        let mut rows = Vec::new();
+    type Reused = Vec<TraceRow<'static>>;
+
+    fn read(&mut self, mut buffer: &[u8], reuse: &mut Self::Reused) -> Result<usize> {
+        reuse.clear();
+        let mut reuse = ClearOnDrop(reuse);
+        // Safety: The Vec is empty and is cleared on leaving this scope, so it's safe to cast the
+        // lifetime of data, since there is no data and data can't leave this function.
+        let rows =
+            unsafe { transmute::<&mut Vec<TraceRow<'_>>, &mut Vec<TraceRow<'_>>>(&mut *reuse) };
         let mut bytes_read = 0;
         loop {
             match postcard::take_from_bytes(buffer) {
```

Comment:
This whole "write rows to a vec" thing makes sense if one of the following is true:

1. You desperately need to reduce the time you hold the store write lock open and can't wait for de-serialization to happen.
2. You don't want partial writes to the store to happen in the case of a deserialization failure.

Number 1 seems like it shouldn't matter because deserialization should be fast. Number 2 might matter, but we can probably assume that deserialization doesn't fail and panic if it does?

If one of those two isn't true, you could call `self.process` inside the deserialization loop, and remove the need for a vec entirely.